src/components/BaseButton/BaseButton.tsx

```diff
@@ -1,20 +1,36 @@
-import React, {FC} from 'react';
+import React, { FC, ComponentProps } from 'react';
+import styles from '../BaseButton/BaseButton.module.scss';
+import cnBind from 'classnames/bind';
 
-export interface BaseButtonProps {
-    children: React.ReactNode;
-    color: string;
+export interface BaseButtonProps extends ComponentProps<'button'> {
+    variant?: 'text' | 'contained' | 'outlined';
+    size?: 'small' | 'medium' | 'large';
 }
 
-const BaseButton: FC<BaseButtonProps> = ({
-                                                 children,
-                                                 color,
-                                                 ...props
-                                             }) => {
+const cx = cnBind.bind(styles);
+
+const BaseButton: FC<BaseButtonProps> = ({ variant, size, ...props }) => {
+    const btn = cx('button', props.className, {
```

Comment:
```suggestion
    const btn = cx('button', props.className, size, variant, {
```
if you pass size and variant like this you want need to specify condition for each size and for each variant
imagine if you will have more 3 variants

Reply:
done